lib/boom_notifier.ex

```diff
@@ -1,16 +1,57 @@
 defmodule BoomNotifier do
   @moduledoc false
+
   # Notify the exception to all the defined notifiers
 
   alias BoomNotifier.ErrorStorage
 
+  def walkthrough_notifiers(settings, callback) do
+    case Keyword.get(settings, :notifiers) do
+      nil ->
+        with {:ok, notifier} <- Keyword.fetch(settings, :notifier),
+             {:ok, options} <- Keyword.fetch(settings, :options) do
+          callback.(notifier, options)
+        end
+
+      notifiers_config when is_list(notifiers_config) ->
+        for notifier_config <- notifiers_config do
+          with {:ok, notifier} <- Keyword.fetch(notifier_config, :notifier),
+               {:ok, options} <- Keyword.fetch(notifier_config, :options) do
+            callback.(notifier, options)
+          end
+        end
+    end
+  end
+
   defmacro __using__(config) do
     quote location: :keep do
       use Plug.ErrorHandler
 
       import BoomNotifier
       require Logger
 
+      settings = unquote(config)
+
+      try do
+        walkthrough_notifiers(
+          settings,
+          &if function_exported?(&1, :validate!, 1) do
+            &1.validate!(&2)
+          end
+        )
+      rescue
+        e ->
+          error_info = Exception.format_banner(:error, e, __STACKTRACE__)
+          [{module, function, arity, _}, {notifier, _, _, _} | _] = __STACKTRACE__
+          Exception.format_mfa(module, function, arity)
```

Comment:
I don't see what happens with the value returned here 🤔

Reply:
I think this was a copy/paste mistake 😅